lib/floe/workflow/runner/docker.rb

```diff
@@ -45,10 +46,66 @@ def cleanup(runner_context)
           delete_secret(secrets_file)    if secrets_file
         end
 
+        def wait(timeout: nil, events: %i[create update delete], &block)
+          until_timestamp = Time.now.utc + timeout if timeout
+
+          r, w = IO.pipe
+
+          pid = AwesomeSpawn.run_detached(
+            self.class::DOCKER_COMMAND, :err => :out, :out => w, :params => wait_params(until_timestamp)
+          )
+
+          w.close
+
+          loop do
+            readable_timeout = until_timestamp - Time.now.utc if until_timestamp
+
+            # Wait for our end of the pipe to be readable and if it didn't timeout
+            # get the events from stdout
+            next if r.wait_readable(readable_timeout).nil?
+
+            # Get all events while the pipe is readable
+            notices = []
+            while r.ready?
+              notice = r.gets
+
+              # If the process has exited `r.gets` returns `nil` and the pipe is
+              # always `ready?`
+              break if notice.nil?
+
+              ref, event, state = parse_notice(notice)
+              next unless events.include?(event)
+
+              state ||= inspect_container(ref)&.dig("State")
+
+              runner_context = {"container_ref" => ref, "container_state" => state}
+
+              notices << runner_context
```

Comment:
Super-minor, but there's no need for a temp var from what I can see
```suggestion
              notices << {"container_ref" => ref, "container_state" => state}
```